File: src/service/mailService.ts
import nodemailer from 'nodemailer';
import { config } from '../config/config';

interface MailData {
  to: string;
  subject: string;
  html: string;
}

const sendMail = async (
  data: MailData,
  callback: (error: Error | null, response: string | null) => void
) => {
  const transporter = nodemailer.createTransport({
    service: 'Gmail',
    auth: {
      user: config.MAIL_ADDRESS,
      pass: config.MAIL_PASSWORD,
    },
  });

  const mailOptions = {
    from: config.MAIL_ADDRESS,
    to: data.to,
    subject: data.subject,
    html: data.html,
  };

  transporter.sendMail(mailOptions, (error, info) => {
    if (error) {
      callback(error, null);
    } else {
      callback(null, info.response);
    }
  });
};

export default sendMail;

File: APIs/app/core/services/mail_service.py
import aiosmtplib
from email.message import EmailMessage
from app.core.config.config import config

async def send_mail(to: str, subject: str, html: str):
    message = EmailMessage()
    message["From"] = config.MAIL_ADDRESS
    message["To"] = to
    message["Subject"] = subject
    message.set_content(html, subtype="html")

    try:
        await aiosmtplib.send(
            message,
            hostname="smtp.gmail.com",
            port=587,
            start_tls=True,
            username=config.MAIL_ADDRESS,
            password=config.MAIL_PASSWORD
        )
        return {"status": "success", "response": "Mail sent successfully"}
    except Exception as e:
        print(f"Error sending mail: {e}")
        # Build callbacks or raise error depending on usage
        return {"status": "error", "message": str(e)}
